app/src/main/java/org/mozilla/tv/firefox/pocket/PocketVideoParser.kt

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.pocket
+
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Handles marshalling [PocketFeedItem.Video] objects from JSON.
+ */
+object PocketVideoParser {
+
+    fun parse(jsonObj: JSONObject): PocketFeedItem.Video? = try {
+        PocketFeedItem.Video(
+            id = jsonObj.getInt("id"),
+            title = jsonObj.getString("title"),
+            // Note that the 'url' property of our domain object can be retrieved from
+            // either of two JSON fields, and we make no distinction internally
+            url = jsonObj.optString("tv_url", null) ?: jsonObj.getString("url"),
+            thumbnailURL = jsonObj.getString("image_src"),
+            popularitySortId = jsonObj.getInt("popularity_sort_id")
+        )
+    } catch (e: JSONException) {
+        null
```

Comment:
This is fragile because if pocket API decides to change JSON key then the users will end up with unhandled state until next app store submission. Generally risky not handling error when parsing raw JSON. Good news is that we are using MVVM. One possible solution is to introduce a `PARSING_ERROR` state

Reply:
We are handling failed parsing, but not in this layer.  In the repo this is treated the same way as any IO exception, which displays either the most recently cached data or a 'try again' button to the user.